internal/scheduler/publisher.go

```diff
@@ -109,7 +110,10 @@ func (p *PulsarPublisher) PublishMessages(ctx context.Context, events []*armadae
 		for _, msg := range msgs {
 			p.producer.SendAsync(sendCtx, msg, func(_ pulsar.MessageID, _ *pulsar.ProducerMessage, err error) {
 				if err != nil {
-					log.WithError(err).Error("error sending message to Pulsar")
+					log.
+						WithField(logging.Stacktrace, logging.ExtractStack(err)).
```

Comment:
Why not use `WithStacktrace`?

Reply:
there is no logrus.withStacktrace

Reply:
But we have a function `logging.WithStacktrace` that does this.

Reply:
You may need to convert to an entry before calling it.